### PR TITLE
2065 mkdir warn popup

### LINKED ISSUE #2065: Add a warning popup for open screenshots or saves folder if they don't exist
At least if the user hasn't taken a screenshot the open screenshots button just silently fails.
Instead it should show a warning popup that the folder doesn't exist and suggest taking a screenshot and then trying again.
The same probably also applies to the browse saves folder button so that should also have a warning popup, and suggesting the player to make a save first.

I think it's fine to have this either way: after closing the warning the folder is attempted to be opened anyway, or nothing is done (trying to open a non-existing folder is not a fatal error so it doesn't matter).

### PR DESCRIPTION
**Pull request by celebrityknave**
**Brief Description of What This PR Does**
This PR adds a pop-up warning if the user attempts to open the screenshots directory from the options menu before having taken any screenshots to create the directory. The same behaviour has also been implemented for opening the saves directory if no saves have been created. A small typo was also fixed in Godot editor, renaming the 'ScreeshotFolder' button.

**Related Issues**
Closes #2065 
-->

**Progress Checklist**

Note: before starting this checklist the issue should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [ ] Initial code review passed
- [ ] Functionality is confirmed working by another person (see above checklist link)
- [ ] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author.

### CODE REVIEW

File: src/general/OptionsMenu.cs [hunk 1]
```diff
@@ -1276,7 +1281,9 @@ private void BuildInputRebindControls()
     private void OnOpenScreenshotFolder()
     {
         GUICommon.Instance.PlayButtonPressSound();
-        OS.ShellOpen(ProjectSettings.GlobalizePath(Constants.SCREENSHOT_FOLDER));
+
+        if (OS.ShellOpen(ProjectSettings.GlobalizePath(Constants.SCREENSHOT_FOLDER)).Equals(Error.FileNotFound))
```
Comment (hhyyrylainen) [inline src/general/OptionsMenu.cs:1285]: Doesn't normal equality comparison `==` work here?

File: src/saving/SaveManagerGUI.cs [hunk 2]
```diff
@@ -194,7 +199,8 @@ private void RefreshButtonPressed()
 
     private void OpenSaveDirectoryPressed()
     {
-        OS.ShellOpen(ProjectSettings.GlobalizePath(Constants.SAVE_FOLDER));
+        if (OS.ShellOpen(ProjectSettings.GlobalizePath(Constants.SAVE_FOLDER)).Equals(Error.FileNotFound))
```
Comment (hhyyrylainen) [inline src/saving/SaveManagerGUI.cs:202]: and here as well

File: src/saving/SaveManagerGUI.tscn [hunk 3]
```diff
@@ -223,6 +224,24 @@ __meta__ = {
...
+margin_right = 1000.0
+margin_bottom = 600.0
+window_title = "NO_SAVE_DIRECTORY"
+
+[node name="DialogText" type="Label" parent="CenterContainer/VBoxContainer/SaveDirectoryWarningDialog"]
+margin_left = 10.0
+margin_top = 10.0
+margin_right = 990.0
+margin_bottom = 32.0
+text = "TRY_NEW_GAME"
```
Comment (hhyyrylainen) [inline src/saving/SaveManagerGUI.tscn:239]: I think this should be named more accurately, something like `TRY_MAKING_A_SAVE`
as that's what creates the folder and not starting a new game...

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]
Thanks for making a PR for Thrive.
Here's a few quick comments from me.

File: src/general/OptionsMenu.cs (hunk 1 above)
Comment (celebrityknave) [reply src/general/OptionsMenu.cs]: No worries. Changed to preferred style.

File: src/saving/SaveManagerGUI.cs (hunk 2 above)
Comment (celebrityknave) [reply src/saving/SaveManagerGUI.cs]: As above.

File: src/saving/SaveManagerGUI.tscn (hunk 3 above)
Comment (celebrityknave) [reply src/saving/SaveManagerGUI.tscn]: No worries. Fixed.

**Review by hhyyrylainen — APPROVED** [review]
Seems to work fine. The texts could have a bit more of an explanation but they are rarely enough seen that I'm not going to ask changes to them at this time.

**Pull request merged**